### PR TITLE
chore: moving off from url-parse

### PR DESCRIPTION
**Pull request by tiwarishubham635**
# Fixes #

Currently, we use the [url-parse](https://www.npmjs.com/package/url-parse) package. This package is soft-deprecated - it's not actually throwing a deprecation warning, but the package's README urges users to switch to the [built-in URL parser](https://nodejs.org/api/url.html) that now exists. The changes have been taken from #998 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-node/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: spec/validation.spec.js [hunk 1]
```diff
@@ -216,7 +216,7 @@ describe("Request validation", () => {
 
   it("should validate urls with special characters", () => {
     const specialRequestUrl = requestUrl + "&Body=It's+amazing";
-    const signature = "dsq4Ehbj6cs+KdTkpF5sSSplOWw=";
```
Comment (manisha1997) [inline spec/validation.spec.js:219]: Why there is a change in signature here?

Comment (tiwarishubham635) [reply spec/validation.spec.js:219]: We had this discussion [here](https://github.com/twilio/twilio-node/pull/998#discussion_r1553172975)

**Review by manisha1997 — APPROVED** [review]

**Comment by tiwarishubham635** [comment]
@43081j I hope this looks good to you. please check. Thanks!

**Review by 43081j — APPROVED** [review]
looks good to me! thanks again for picking it up 🙏

**Pull request merged**